zookeeper/core/component.py

```diff
@@ -572,6 +572,42 @@ def configure(
         # track of that fact.
         instance.__component_fields_with_values_in_scope__.add(field.name)
 
+    # Check that all `conf` values are being used, and throw if we've been
+    # passed an un-used option.
+    for key in conf:
+        error = ValueError(
+            f"Key '{key}' does not correspond to any field of component "
+            f"'{instance.__component_name__}'."
+            "\n\n"
+            "If you have nested components as follows:\n\n"
+            "```\n"
+            "@component\n"
+            "class ChildComponent:\n"
+            "    a: int = Field(0)\n"
+            "\n"
+            "@task\n"
+            "class SomeTask:\n"
+            "    child: ChildComponent = ComponentField(ChildComponent)\n"
+            "    def run(self):\n"
+            "        print(self.child.a)\n"
+            "```\n\n"
+            "then trying to configure `a=<SOME_VALUE>` will fail. You instead need to "
+            "fully qualify the key name, and configure the value with "
+            "`child.a=<SOME_VALUE>`."
+        )
+
+        if "." in key:
+            scoped_component_name = key.split(".")[0]
+            if not (
+                scoped_component_name in instance.__component_fields__
+                and isinstance(
+                    instance.__component_fields__[scoped_component_name], ComponentField
+                )
```

Comment:
Let's say the key is `"<x>.<y>"` where `<x>` does not contain a full stop.

If `<x>` is not a valid field name then this will trigger the if statement through line 602.

If `<x>` is a field name, corresponding to a component field, then the check here will pass, but when the sub-component is being recursively configured there will be a check for `<y>` being a valid field name, et cetera.